src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -299,33 +292,9 @@ private DataElement CreateDataElement(HttpRequest request, string elementType, I
                 contentType = request.ContentType;
             }
 
-            string dataId = Guid.NewGuid().ToString();
-
-            string dataLink = $"{prefix}/instances/{instance.Id}/data/{dataId}";
-
-            // create new data element, store data in blob
-            DataElement newData = new DataElement
-            {
-                // update data record
-                Id = dataId,
-                ElementType = elementType,
-                ContentType = contentType,
-                CreatedBy = User.Identity.Name,
-                CreatedDateTime = creationTime,
-                FileName = contentFileName ?? $"{dataId}.xml",
-                LastChangedBy = User.Identity.Name,
-                LastChangedDateTime = creationTime,
-
-                DataLinks = new ResourceLinks()
-                {
-                    Apps = dataLink,
-                },
+            string userName = GetNameOfUser(User);
```

Comment:
Should be userId, the user id number not name.

Reply:
Same as above.